pkg/reconcilermanager/controllers/reconciler_base.go

```diff
@@ -194,8 +194,21 @@ func (r *reconcilerBase) createOrPatchDeployment(ctx context.Context, declared *
 		return controllerutil.OperationResultCreated, nil
 	}
 
-	// TODO: check if VPA is enabled
-	vpaEnabled := false
+	//check if allow vertical scale is enabled
+	allowVerticalScale := false
```

Comment:
You should be able to get the setting from the reconciler flag var `allowVerticalScale` and pass it here, instead of making an API to get the information from the cluster.

Reply:
as per previous comment, allowVerticalScale will be passed using reconcilerBase